thelper/nn/segmentation/fcn.py

```diff
@@ -0,0 +1,63 @@
+import torch
+import torchvision
+import thelper.nn
+
+
+class fcn_resnet50(thelper.nn.utils.Module):
```

Comment:
Since the procedure is the same between all variants, it should be a base class. 
From a quick scan, the only difference seems to be the `self.model = ...` variation.

Make a `DeeplablV3ResNetBase` and all of these should use it like so:

```
class SegmResNetBase(thelper.nn.utils.Module):
    def __init__(self, task, pretrained=False):
        self.model = None
        [... all other operations ....]
    [... all other methods....]

class FCNResNet50(SegmResNetBase):
    def __init__(self, task, pretrained=False):
        self.model = torchvision.models.segmentation.fcn_resnet101(pretrained=self.pretrained)
        super(FCNResNet50, self).__init__(task, pretrained=pretrained)
```

Adjust other parameters similarly if needed.